src/lib/utils/size_estimation_utils.hpp

```diff
@@ -17,7 +17,8 @@ size_t
 string_heap_size(const T& string) {
   // Get the default pre-allocated capacity of SSO strings. Note that the empty string has an unspecified capacity, so
   // we use a really short one here.
-  const auto sso_string_capacity = T{"."}.capacity();
```

Comment:
Just to get you right. The previous code here `T{"."}.capacity();` compiles just fine but one assert does fail in that case?

Reply:
Yes!
Do you have an idea how this could be?

Reply:
Tried the following for `std::string`:
```c++
const auto sso_string_capacity = std::string{"."}.capacity();
const auto sso_string = std::string{"."};
const auto sso_string_capacity_2 = sso_string.capacity();
std::cout << sso_string_capacity << " " << sso_string_capacity_2 << std::endl;
```
and it outputs (as expected)
```
22 22
```
for both clang-14 and clang-15 on macOS. Seems to be a quite weird error.

Reply:
How does the (unedited) test case behave on your machine?

Reply:
Tried it and it's just getting weirder. I added the following print:
```c++
const auto sso_string_capacity = T{"."}.capacity();
const auto small_string = T{"."};
const auto sso_string_capacity_2 = small_string.capacity();
std::cout << string.capacity() << " " << sso_string_capacity << " " << sso_string_capacity_2 << " "
          << T{"."}.capacity() << " " << small_string.capacity() << std::endl;
```
and it outputs
```
22 23 22 22 22
```

I guess adding a comment here would be helpful. Furthermore, I wonder if we can capture the SSO capacity as a constant to avoid the evaluation for each execution of the utility function.

Reply:
From my understanding `std::string.capacity()` should always return the same value for current implementations using sso for short strings: I found something like the following implementation for short strings: `return static_cast<size_type>(__min_cap)) - 1;`
My guess, the weird 23 comes from some compiler magic/optimization, because the capacity could be determined at compile time. I am not sure whether the current fix handles future compiler optimizations well (better than the previous version).

Reply:
This is why, removing the assert and leaving a comment is maybe a better fix.
Just as note: besides evaluating the string's capacity to assume sso, one could also evaluate the address of data(). (see, for example, https://stackoverflow.com/questions/65736613/c-sso-how-to-programatically-find-if-a-stdwstring-is-allocated-with-short)